examples/blurredbox/shaders/blurredbox.frag

```diff
@@ -0,0 +1,84 @@
+// opacity of the rectangle
+uniform lowp float rectOpacity;
+// x = top left, y = top right, z = bottom right, w = bottom left
+uniform lowp vec4 rectCornerRadii;
+// the rectangle on the screen (x, y, width, height)
+uniform lowp vec4 rectOnScreen;
+// the screen's rectangle (x, y, width, height)
+uniform lowp vec4 rectOfScreen;
+// must be greater than 0.0!
+uniform lowp float blurDirections;
+// must be greater than 0.0!
+uniform lowp float blurQuality;
+// the radius for blurring
+uniform lowp float blurSize;
+
+// normalized position of the fragment
+varying lowp vec2 coord;
+
+// the texture
+uniform sampler2D txr;
+
+lowp float effectiveRadius( in lowp vec4 radii, in lowp vec2 point )
+{    
+    // aliases
+    lowp float px = point.x * rectOfScreen.z;
+    lowp float py = point.y * rectOfScreen.w;
+    lowp float rx = rectOnScreen.x;
+    lowp float ry = rectOnScreen.y;
+    lowp float rw = rectOnScreen.z;
+    lowp float rh = rectOnScreen.w;
+
+    // predicates
+    bool l = px >= rx && px < rx + rw / 2.0;
+    bool r = px >= rx + rw / 2.0 && px < rx + rw;
+    bool t = py >= ry && py < ry + rh / 2.0;
+    bool b = py >= ry + rh / 2.0 && py < ry + rh;
+
+    if ( t && l) return radii.x;
+    if ( t && r) return radii.y;
+    if ( b && r) return radii.z;
+    if ( b && l) return radii.w;
+    return 0.0;
+}
+
+// from https://iquilezles.org/articles/distfunctions
+lowp float roundedBoxSDF(vec2 centerPosition, vec2 size, lowp float radius)
+{
+    return length( max( abs(centerPosition) - size + radius, 0.0) ) - radius;
+}
+
+// source: https://www.shadertoy.com/view/Xltfzj
+void main()
+{
+    if(rectOpacity == 0.0)
+    {
+        return;
+    }
+
+    // Radius for blurring around the pixel
+    vec2 blurRadius = blurSize / rectOnScreen.zw;
+
+    // Pixel colour
+    vec4 fragColor = texture2D(txr, coord);
+
+    // Blur calculations
+    const lowp float Pi = 6.28318530718; // constant for Pi*2
+    for( float d = 0.0; d < Pi; d += Pi / blurDirections)
+    {
+        for(float i = 1.0 / blurQuality; i <= 1.0; i += 1.0 / blurQuality)
+        {
+            fragColor += texture2D( txr, coord + vec2(cos(d), sin(d)) * blurRadius * i);
```

Comment:
cos/sin can be calculated outside of the inner loop. Actually it would good enough to calculate only cos and use Pythagoras for the sin ( or v.v. ) then. Can imagine that this will speed up the shader significantly. Working with precalculated tables for certain values of blurDirections should be make things even faster.

This shader also seems to do all the expensive blur calculations for pixels that will be sorted out later.

But of course things have to work first before doing performance improvements,